### PR TITLE
Edit wizard directly on dashboard

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description
Edit wizard directly on dashboard following the steps:
1. choose a dashboard
2. click `Edit` button on the top nav bar of the dashboard
3. select the setting button on the top right of a wizard on that dashboard
4. click edit wizard

Screen Shot 2022-10-05 at 10 49 47 AM

When finish editing wizard, there are some different options for saving that wizard, this PR ensures the below five different flows for editing wizard all work as expected:
1. `Save and return` button on top nav bar --> will save the original wizard and return back to the dashboard without displaying an additional save modal
2. `Save as` button on top nav bar --> will open a save modal
   * `save as new wizard` toggle on + `add to dashboard after saving` toggle on + Save and return button
   * `save as new wizard` toggle on + `add to dashboard after saving` toggle off + Save button
   * `save as new wizard` toggle off + `return to dashboard after saving` toggle on + Save and return button
   * `save as new wizard` toggle off + `return to dashboard after saving` toggle off + Save button
 
Screen Shot 2022-10-05 at 10 58 12 AM

Screen Shot 2022-10-05 at 10 57 32 AM

Screen Shot 2022-10-05 at 10 57 20 AM

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2382
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
Now that we have some test coverage for saving and loading (thanks!), we should also include unit tests for this flow, particularly because there are five separate user flows that are a little confusing to mentally model.

**Comment by abbyhu2000** [comment]
> Now that we have some test coverage for saving and loading (thanks!), we should also include unit tests for this flow, particularly because there are five separate user flows that are a little confusing to mentally model.

@joshuarrrr Yea i agree. Do you think it is more appropriate to write functional tests for these flows? Since it involves with user clicking different button and toggle. 

Also when i was writing the unit tests for get_top_nav_config() function, i included one unit test for the edit logic there. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b75e07dd45d4d6b189fbd177ee5bad60a1d08505/src/plugins/wizard/public/application/utils/get_top_nav_config.test.tsx#L140

**Comment by ashwin-pc** [comment]
@joshuarrrr yeah there is a functional test we plan on adding for this flow.

**Review by ashwin-pc — DISMISSED** [review]

**Comment by ashwin-pc** [comment]
> > Now that we have some test coverage for saving and loading (thanks!), we should also include unit tests for this flow, particularly because there are five separate user flows that are a little confusing to mentally model.
> 
> @joshuarrrr Yea i agree. Do you think it is more appropriate to write functional tests for these flows? Since it involves with user clicking different button and toggle.
> 
> Also when i was writing the unit tests for get_top_nav_config() function, i included a unit test for the edit logic there.
> 
> https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b75e07dd45d4d6b189fbd177ee5bad60a1d08505/src/plugins/wizard/public/application/utils/get_top_nav_config.test.tsx#L140

@abbyhu2000 how was that test for the edit flow previously passing if the change is being made here?

**Comment by abbyhu2000** [comment]
> > > Now that we have some test coverage for saving and loading (thanks!), we should also include unit tests for this flow, particularly because there are five separate user flows that are a little confusing to mentally model.
> > 
> > 
> > @joshuarrrr Yea i agree. Do you think it is more appropriate to write functional tests for these flows? Since it involves with user clicking different button and toggle.
> > Also when i was writing the unit tests for get_top_nav_config() function, i included a unit test for the edit logic there.
> > https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b75e07dd45d4d6b189fbd177ee5bad60a1d08505/src/plugins/wizard/public/application/utils/get_top_nav_config.test.tsx#L140
> 
> @abbyhu2000 how was that test for the edit flow previously passing if the change is being made here?

@ashwin-pc When i am writing my last PR(create wizard directly from dashboard), i include the editing logic in the onSave() function. So when i am writing unit tests for [onSave()](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/ac07159c6e4e9444bb5f20ce54cdf4e42d219038/src/plugins/wizard/public/application/utils/get_top_nav_config.tsx#L115), i am able to mock the props of onSave and test that part of the logic. However, in order for editing wizard from dashboard to work, the logic that's needed inside the [topNavConfig](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/ac07159c6e4e9444bb5f20ce54cdf4e42d219038/src/plugins/wizard/public/application/utils/get_top_nav_config.tsx#L67) constant to call that onSave() function with the correct props are still missing, so this PR added those logics.

**Review by ananzh — DISMISSED** [review]
LGTM

**Review by ananzh — APPROVED** [review]

File: src/plugins/wizard/public/application/utils/get_top_nav_config.tsx [hunk 1]
```diff
@@ -100,6 +105,46 @@ export const getTopNavConfig = (
...
+            }),
+            emphasize: true,
+            iconType: 'checkInCircleFilled' as const,
+            description: i18n.translate(
+              'wizard.topNavMenu.saveAndReturnVisualizationButtonAriaLabel',
+              {
+                defaultMessage: 'Finish editing wizard and return to the last app',
+              }
+            ),
+            testId: 'wizardsaveAndReturnButton',
```
Comment (kavilla) [inline src/plugins/wizard/public/application/utils/get_top_nav_config.tsx:123]: NIT: Casing is off on this one.

**Review by kavilla — APPROVED** [review]

**Pull request merged**